runtime/doc/options.txt

```diff
@@ -2133,44 +2124,13 @@ A jump table for the options with a short description can be found at |Q_op|.
 'edcompatible' 'ed'	Removed. |vim-differences| {Nvim}
 
 					*'encoding'* *'enc'* *E543*
-'encoding' 'enc'	string (default: "utf-8")
-			global
-			{only available when compiled with the |+multi_byte|
-			feature}
-	Sets the character encoding used inside Vim.  It applies to text in
-	the buffers, registers, Strings in expressions, text stored in the
-	shada file, etc.  It sets the kind of characters which Vim can work
-	with.  See |encoding-names| for the possible values.
+'encoding' 'enc'	Removed. |vim-differences| {Nvim}
+	Nvim always uses UTF-8 internally.
```

Comment:
I think it's worth keeping the mention of RPC/remote as well:

    Nvim always uses UTF-8 internally. RPC communication 
    (remote plugins/GUIs) must use UTF-8 strings.